### ISSUE TITLE
You're lacking Level 1 Skills.

### ISSUE THREAD

**Issue by BenjixD** [ronancpl/MapleSolaxiaV2#114]
Hey again,
Benjamin here from DietStory; adaptation of MapleSolaxia. 

We're currently in the midst of re-creating Ultimate Adventurers to promote Cygnus Knights gameplay. We've come to the final point where our UAs are available to obtain their respective Cygnus Knight signature skills (ie. <Soul Driver> for Warriors). 

Now long story short, because v83 does not have the UA skill adaptations, we decided just to give each UA the level 20/30 versions of the CK skills via teachSkill(). Here is where the problem happens:

The game does not let UAs add skill points (if they are under a certain level) into their 2nd/3rd job skills because the Cygnus Knight skills are "taking away" the usable SPs and rendering the remaining unusable.

Example:
Suppose I am a level 50 UA Assassin, with Vampire at level 20 (because it was given).
I have 61 SP in my Skill tab (I've applied all 1st job SP), but I can only assign 41 of them. As you can imagine, the remaining 20 SP cannot be assigned into 2nd job anymore due to the client message "You're lacking Level 1 Skills." From what we can tell, its because the game checks how many skill points have been assigned to 2nd job and will not let you use more than 61 at this level.

I know this is likely a client-sided issue and may require client modding (or we add a low-key command line SP setter which isn't the best UX), but curious how many Rebirth servers get around this issue. 

Also I'd like to extend my thanks on behalf of our small team for your work on MapleSolaxia and hope we can continue to share and perhaps collaboratively work on new content.

TL;DR 
Giving skills to characters locks them from raising some SP into their job and thanks for all the hard work :)

**Comment by ronancpl** [comment]
Regarding this 2nd job SP locking issue, I suggest checking the player's SP data on the DB. This locking usually occurs when all the SP for the skill tier or greater have been used up already, and you start trying to apply lower tier SP into the skill.

In other words, debug whatever is happening with the player's SP structure that doesn't let you starting applying SP on 2nd job skills.

**Comment by ronancpl** [comment]
Do you have disc Benji?

**Comment by BenjixD** [comment]
Hey Ronan,

I suppose Disc as in Discord? Yes I'd like to connect sometime.

As for what I've debugged so far, I am arguably sure this is a client issue. 

Jumping back to the LVL 50 assassin with vampire example, lets suppose I haven't raised any 2nd Job skill points and have raised all my 61 SP points for 1st Job.

I have now:
- 61 SP usable for 2nd Job : [(50 - 30) * 3 + 1]
- I can apply 41 of those points into anything, ie Claw Mastery
- The remaining 20, can only be applied to 1st job skills
(Technically, 61 of my 2nd job SP have been used since I've used 41 + 20 from vampire)

### Test 1 - Give the player more SP:
1. using the SP command, gave myself 100 SP instead of 61
- Doesn't work, can only apply 41, the remaining 59 can only be applied to 1st Job
2. Level myself up to 51
- I can now apply 44 SP, but the remaining 20 still cannot be used to 2nd Job

### Test 2 - Through the DB, max the Night Walker 1st Job skills for my Assassin
1. Gave NightWalker's Dark Sight 61 skill points (I'm confident I don't need to do a true distribution as long as the skills add to 61)
- Results in the same error, cannot raise over 41 points to 2nd Job at level 50

**So what does this mean**
My assumption is, the client forbids you from putting more skill points into your corresponding job level in relation to your level. 
- Level 50 - (61 SP to 2nd Job only)
- Level 51 - (64 SP to 2nd Job only)
...
- Level 70 - (121 SP to 2nd Job only) / (1 SP to 3rd Job only)
- Level 71 - (124 SP to 2nd Job only <unsure>) / (4 SP to 3rd Job only)
...
- Level 120+ - (No bars held probably)

If my assumption is correct, then we'll need to do client modding to solve this issue. However, if this is the case, how do rebirthing servers get around this issue? Perhaps maybe you level too fast to even notice!

And so our team is working on a command-line version to add skill points which bypasses the client check. Of course I'd really love to avoid such complications for players.

**Comment by resinate** [comment]
the client i released on ragezone fixes this SP issue
